keras/src/ops/nn_test.py

```diff
@@ -1248,6 +1250,17 @@ def test_softmax(self):
             ],
         )
 
+    def test_softmax_correctness_with_axis_tuple(self):
+        if backend.backend() == "torch":
+            pytest.skip("unsupported tuple dim by torch")
```

Comment:
It's not supported how? Could we support it? The code is written in backend-agnostic ops.

Reply:
First of all, the reason why I skipped this test case for torch is related in below code.
When calling torch.nn.softmax, it calls torch.nn.functional.softmax, and this function is declared with dim param with Optional[int].

```
# torch.nn.py
import torch.nn.functional as tnn

def softmax(x, axis=-1):
     ...
    else:
        output = tnn.softmax(x, dim=axis)
```

```
# torch.nn.functional.py

def softmax(input: Tensor, dim: Optional[int] = None, _stacklevel: int = 3, dtype: Optional[DType] = None) -> Tensor:
```

Because of this, I got an error like this.
```
TypeError: softmax() received an invalid combination of arguments - got (tuple), but expected one of:
 * (int dim, torch.dtype dtype)
 * (name dim, *, torch.dtype dtype)
```

Actually, on the documentation for softmax in keras, axis argument is described as Integer type, so, I'm not sure this code for supporting tuple axis is needed. 

```
@keras_export(["keras.ops.softmax", "keras.ops.nn.softmax"])
def softmax(x, axis=-1):
    ...
    Args:
        x: Input tensor.
        axis: Integer, axis along which the softmax is applied.

    Returns:
        A tensor with the same shape as `x`.
```

Reply:
So, As I think, there could be 2 options to keep consistency with all backend types [ tensorflow, torch, jax ]

1. Support tuple axis for softmax on torch, and, update the document to allow tuple type for axis argument also.

2. Block to call softmax function with tuple axis on tensorflow and jax also.

Reply:
Oh, sorry. I was little bit confused when writing test case. I correct my test case to call keras.ops.nn.softmax, and it works well now. Thanks.